client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -254,38 +258,42 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder) {
 
     @Override
     public CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
+                                                  final Map<Double, Double> newKeyRanges, final long timeoutMillis,
```

Comment:
For a long poll, it is fine if the request errors and the app needs to send it again. But, let's go back to the real issue I have. I'd rather not have this timeout in the API, it is not good, and ideally not the executor either. I'm not sure whether you can remove the executor without creating another one internally, which is not good either.